### PR TITLE
[update] STM32MP1-DK1 BSP drivers

### PR DESCRIPTION
**Pull request by thread-liu**
## 拉取/合并请求描述：(PR description)

[
按照 ST 要求， 更新 BSP 驱动。本次更新已在 DK1 开发板上测试完成。
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

File: bsp/stm32/libraries/STM32MPxx_HAL/CMSIS/Device/ST/STM32MP1xx/Source/Templates/iar/startup_stm32mp15xx.s [hunk 1]
```diff
@@ -225,7 +225,7 @@ __vector_table
 	 SECTION .text:CODE:NOROOT:REORDER(2) 
 Reset_Handler 
 
-          LDR     R0, =SystemInit 
+                      LDR     R0, =SystemInit 
           BLX     R0
```
Comment (BernardXiong) [inline bsp/stm32/libraries/STM32MPxx_HAL/CMSIS/Device/ST/STM32MP1xx/Source/Templates/iar/startup_stm32mp15xx.s:229]: Please take care of indentation.

File: bsp/stm32/libraries/STM32MPxx_HAL/STM32MP1xx_HAL_Driver/Inc/stm32mp1xx_hal.h [hunk 2]
```diff
@@ -28,6 +28,7 @@
 
 /* Includes ------------------------------------------------------------------*/
 #include "stm32mp1xx_hal_conf.h"
+#include <rtthread.h>
```
Comment (BernardXiong) [inline bsp/stm32/libraries/STM32MPxx_HAL/STM32MP1xx_HAL_Driver/Inc/stm32mp1xx_hal.h:31]: Should we modify HAL code to add RT-Thread header file? Please check the reason.

File: bsp/stm32/stm32mp157a-st-discovery/board/ports/timer_sample.c [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    /* 以读写方式打开设备 */
+    ret = rt_device_open(hw_dev, RT_DEVICE_OFLAG_RDWR);
+    if (ret != RT_EOK)
+    {
+        rt_kprintf("open %s device failed!\n", HWTIMER_DEV_NAME);
+        return ret;
+    }
+
+    /* 设置超时回调函数 */
```
Comment (BernardXiong) [inline bsp/stm32/stm32mp157a-st-discovery/board/ports/timer_sample.c:69]: Please add comments in English.

File: bsp/stm32/libraries/STM32MPxx_HAL/STM32MP1xx_HAL_Driver/Inc/stm32mp1xx_hal.h (hunk 2 above)
Comment (thread-liu) [reply bsp/stm32/libraries/STM32MPxx_HAL/STM32MP1xx_HAL_Driver/Inc/stm32mp1xx_hal.h:31]: Yes, RT-Thread header file needs to be added for each HAL library in the STM32 series. If not do that, an error is reported when using the command ` scons -jx` : NULL is undefined.

**Pull request closed without merging**